File: pmod_swt/pmod_swt.c
#include <stdio.h>
#include <wiringPi.h>
#include <string.h>

#define SWITCH_COUNT 8 // Nombre d'interrupteurs
int switch_pins[SWITCH_COUNT] = {8, 10, 9, 11, 19, 21, 20, 18};

int main(int argc, char *argv[]) {
    int switch_state = 0;

    if (argc > 1 && strcmp(argv[1], "JA") == 0) {
        printf("Port JA sélectionné !\n");
    } else {
        printf("Port inconnu : %s\n", argv[1]);
    }

    // Initialisation de WiringPi
    if (wiringPiSetupGpio() == -1) { // Utilisation de la numérotation BCM
        perror("Erreur : impossible d'initialiser WiringPi");
        return 1;
    }

    // Configuration des broches en entrée
    for (int i = 0; i < SWITCH_COUNT; i++) {
        pinMode(switch_pins[i], INPUT);
        pullUpDnControl(switch_pins[i], PUD_DOWN); // Activation de la résistance pull-down
    }

    // Lire l'état des interrupteurs
    for (int i = 0; i < SWITCH_COUNT; i++) {
        if (digitalRead(switch_pins[i]) == HIGH) {
            switch_state |= (1 << i);
        }
    }

    printf("État des interrupteurs : 0x%X\n", switch_state);
    
    return switch_state;
}

File: pmod_swt/pmod_swt.py
import wiringpi
import sys

# Liste des broches en mode BCM
switch_pins = [8, 10, 9, 11, 19, 21, 20, 18]

# Initialisation de WiringPi en mode BCM (Broadcom)
wiringpi.wiringPiSetupGpio()

# Configurer les broches en entrée
for pin in switch_pins:
    wiringpi.pinMode(pin, 0)  # 0 = mode INPUT
    wiringpi.pullUpDnControl(pin, 0)  # 0 = Pas de résistance interne

def read_switches():
    """Lit l'état des interrupteurs et retourne la valeur en hexadécimal."""
    switch_state = sum(1 << i for i, pin in enumerate(switch_pins) if wiringpi.digitalRead(pin))
    return switch_state

if __name__ == "__main__":
    port = sys.argv[1] if len(sys.argv) > 1 else "Inconnu"

    if port == "JA":
        print("Port JA sélectionné !")
    else:
        print(f"Port inconnu : {port}")

    switch_state = read_switches()
    print(f"État des interrupteurs : 0x{switch_state:X}")

    sys.exit()
